modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotCheckProcess.java

```diff
@@ -0,0 +1,485 @@
+                    kctx.cluster().get().localNode().consistentId())
+            );
+
+            if (!F.isEmpty(metasRes.exceptions()))
+                throw new IgniteSnapshotVerifyException(metasRes.exceptions());
+        }
+        catch (Exception e) {
+            contexts.remove(snpName);
+
+            if (clusterOpFut != null)
+                clusterOpFut.onDone(e);
+
+            return;
+        }
+
+        List<SnapshotMetadata> locMetas = metas.get(kctx.cluster().get().localNode());
+
+        ctx.locMeta = F.isEmpty(locMetas) ? null : locMetas.get(0);
+
+        if (clusterOpFut != null)
+            ctx.clusterMetas = metas;
+
+        if (U.isLocalNodeCoordinator(kctx.discovery()))
+            phase2PartsHashes.start(reqId, ctx.req);
+    }
+
+    /** Finds current snapshot name from the metas. */
+    private @Nullable String snpName(@Nullable Map<UUID, CheckResultDTO> results) {
+        if (F.isEmpty(results))
+            return null;
+
+        for (CheckResultDTO nodeRes : results.values()) {
+            if (nodeRes == null || F.isEmpty(nodeRes.metas))
+                continue;
+
+            assert nodeRes.metas.get(0) != null : "Empty snapshot metadata in the results";
+            assert !F.isEmpty(nodeRes.metas.get(0).snapshotName()) : "Empty snapshot name in a snapshot metadata.";
+
+            return nodeRes.metas.get(0).snapshotName();
+        }
+
+        return null;
+    }
+
+    /** Starts the snapshot full validation. */
+    public IgniteInternalFuture<SnapshotPartitionsVerifyTaskResult> start(
+        String snpName,
+        @Nullable String snpPath,
+        @Nullable Collection<String> grpNames,
+        boolean inclCstHndlrs
+    ) {
+        assert !F.isEmpty(snpName);
+
+        UUID reqId = UUID.randomUUID();
+
+        List<UUID> requiredNodes = new ArrayList<>(F.viewReadOnly(kctx.discovery().discoCache().aliveBaselineNodes(), F.node2id()));
+
+        // Initiator is also a required node. It collects the final oparation result.
+        requiredNodes.add(kctx.localNodeId());
+
+        SnapshotCheckProcessRequest req = new SnapshotCheckProcessRequest(
+            reqId,
+            requiredNodes,
+            snpName,
+            snpPath,
+            grpNames,
+            0,
+            inclCstHndlrs
+        );
+
+        GridFutureAdapter<SnapshotPartitionsVerifyTaskResult> clusterOpFut = new GridFutureAdapter<>();
+
+        clusterOpFut.listen(fut -> {
+            clusterOpFuts.remove(reqId);
+
+            if (log.isInfoEnabled())
+                log.info("Finished snapshot checking process [req=" + req + ']');
+        });
+
+        clusterOpFuts.put(reqId, clusterOpFut);
+
+        phase1CheckMetas.start(req.requestId(), req);
+
+        return clusterOpFut;
+    }
+
+    /** Properly sets errror to the cluster operation future. */
+    static boolean finishClusterFutureWithErr(
```

Comment:
This method invokes the only branch, as `propogatedError == null`

Reply:
No. In `SnapshotManager` the error might appear.

Reply:
This method defined in this class, and it's used in single pattern here. It looks like `IgniteSnapshotManager` needs own method for its specific case. Also holding in mind that `IgniteSnapshotManager` will not check anything in the future and delegate it to `SnapshotCheckProcess`.